skglm/estimators.py

```diff
@@ -890,7 +899,14 @@ def path(self, X, y, alphas, coef_init=None, return_n_iter=True, **params):
             The number of iterations along the path. If return_n_iter is set to
             ``True``.
         """
-        penalty = compiled_clone(MCPenalty(self.alpha, self.gamma))
+        weights = np.ones(X.shape[1]) if self.weights is None else self.weights
```

Comment:
This way we would use `WeightedMCP` even though the `self.weights` isn't provided.
We better use the same logic as you implemented in `.fit` method.

Reply:
done thx!